mautic/mautic
# New post save api hook

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Enhancement
| Automated tests included? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This hook will allow plugins to get the saved entity and the whole Request and do their stuff with it. It will allow them to send more data for example with the contact create request and connect those to the contact.

This also fixes one bug where API contact updates threw a lot of PHP notices as the field group `all` is artificially added to the field groups and has different array schema. So the changes in this PR ignores the `all` group.

#### Steps to test this PR:
1. There is nothing to test as this is not adding any behavior change. Code review should suffice.
2. If someone wants to ensure that API still works, do that.
